internal/generator/vector/output/common/secret_reader.go

```diff
@@ -0,0 +1,28 @@
+package common
+
+import (
+	"fmt"
+	"github.com/openshift/cluster-logging-operator/internal/collector/common"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+	corev1 "k8s.io/api/core/v1"
+	"strings"
+)
+
+const scriptTmpl = ` "%s_%s":{"value":"$(cat %s)","error":null}`
+
+func GenerateSecretReaderScript(secrets []corev1.Secret) string {
+	var scriptBuilder strings.Builder
+	scriptBuilder.WriteString("#!/bin/bash\ncat <<EOF\n{\n")
+	var values []string
+	for _, secret := range secrets {
+		for i, _ := range secret.Data {
+			values = append(values, fmt.Sprintf(scriptTmpl,
+				helpers.Replacer.Replace(secret.Name),
+				helpers.Replacer.Replace(i),
+				common.SecretPath(secret.Name, i)))
```

Comment:
This has to also include the key name to get the value of the actual file.  The mount root will be the secret name but there are keys underneath that are specific values.  Maybe changing the "i" to "key" will make it more obvious what we are doing